proto/cosmos/base/reflection/v1beta1/reflection.proto

```diff
@@ -21,6 +25,187 @@ service ReflectionService {
   };
 }
 
+// AppDescriptor describes a cosmos-sdk based application
+message AppDescriptor {
+  // chain provides the chain descriptor
+  ChainDescriptor chain = 2;
+  // codec provides metadata information regarding codec related types
+  CodecDescriptor codec = 3;
+  // configuration provides metadata information regarding the sdk.Config type
+  ConfigurationDescriptor configuration = 4;
+  // query_services provides metadata information regarding the available queriable endpoints
+  QueryServicesDescriptor query_services = 5;
+  // tx provides metadata information regarding how to send transactions to the given application
+  TxDescriptor tx = 6;
+}
+
+// TxDescriptor describes the accepted transaction type
+message TxDescriptor {
+  // fullname is the protobuf fullname of the raw transaction type (for instance the tx.Tx type)
+  string fullname = 1;
```

Comment:
Maybe this is okay but I kind of have mixed feelings about explicitly supporting non-standard transaction types. As we were going through the Stargate migration, it was my feeling (and I believe other's as well) that making `sdk.Tx` an interface and supporting multiple transaction types in the ante handler layer is something that adds a lot of overhead without adding a lot of value. I'm not saying I'm totally opposed to keeping things as is and allowing multiple Tx types and describing that with reflection here BUT it does feel counter to a lot of the learnings from Stargate. In particular, if chains support different Tx types all sorts of things like block explores and wallets become incompatible between Cosmos chains. I'd personally rather see us standardize on one `Tx` type and not support polymorphism here.

Reply:
I know, this is not intended to be used to support polymorphysm, it's meant to signal to reflection clients if they can write on the chain. 
Ex: if we make a breaking change and decide to switch the tx.Tx object, then a reflection client would know it cannot write to this chain because it can support `cosmos.base.v1beta1.Tx` but not `cosmos.base.v2beta1.Tx`